File: concept/rfid.js
var SerialPort = require("serialport").SerialPort
var serialPort = new SerialPort("/dev/ttyAMA0", {
	baudrate: 9600
});

serialPort.on("open", function () {
	console.log('open');
	serialPort.on('data', function(data) {
		var charCodes = unpack(data),
			cardID = "";

		for(var i = 0, len = charCodes.length; i<len; i++){
			var h = charCodes[i].toString(16);
			cardID += (h.length === 1 ? "0" : "") + h;
		}

		console.log('card id: ' + cardID);
	});  

	serialPort.write(0x02, function(err, results) {
		if(err){
			console.log("Error while writing", err);
		}
	});  
});


function unpack(str) {
    var bytes = [];
    for(var i = 0; i < str.length; i++) {
        var char = str[i];
        bytes.push(char & 0xFF);
    }
    return bytes;
}

File: concept/rfid.py
import serial
import binascii
import sys
import struct

ser = serial.Serial('/dev/ttyAMA0', 9600, timeout=1)

while True:
    ser.write('\x02')
    while(1):
        buf = ser.read(4)
        if len(buf) > 0:
	    print "Serial #:" ,

	    for i in buf:
	        value = struct.unpack('B', i)[0]
	        print "%02x" % (value),

            print ""

    string = ser.read(12)

    if len(string) == 0:
        print "Please insert a tag"
        continue
    else:



	print string.decode('utf-8')
	string = string[1:11] #exclude start x0A and stop x0D bytes

        if string == '0415DB18A3': 
            print "You used your black tag"
        elif string == '0F03028F57':
            print "You used your white tag"
        else:
            print "You do not have a valid tag"
